src/root/rmem.d

```diff
@@ -127,7 +127,7 @@ else
     __gshared size_t heapleft = 0;
     __gshared void* heapp;
 
-    extern (C++) void* allocmemory(size_t m_size)
+    extern (C) void* allocmemory(size_t m_size)
```

Comment:
Yes, it should be used everywhere in this pull request.  You've already used it in several places in this PR where is isn't strictly required, I don't see what the problem is.